lib/event_listeners.js

```diff
@@ -264,7 +277,11 @@ AWS.EventListeners = {
         });
 
         httpResp.on('end', function onEnd() {
-          if (!stream || !stream.didCallback) {
+          if ((AWS.HttpClient.streamsApiVersion !== 2) ||
+            // don't concat response chunks when using event streams with streams unless response is unsuccessful
+            (!(operation.hasEventOutput && service.successfulResponse(resp)) &&
+            (!stream || !stream.didCallback))
```

Comment:
There's no issue here but a little hard to follow.  Will using another if statement be more readable? like this:

```javascript
if (!stream || !stream.didCallback) {
//don't concat response chunks when using event streams unless response is unsuccessful
    if ((AWS.HttpClient.streamsApiVersion === 2) && operation.hasEventOutput && service.successfulResponse(resp) ) {
         return
    }
   resp.request.emit('httpDone');
    done();
}
```
I think it's more readable(?) It's your call.